astropy/io/ascii/tests/test_write.py

```diff
@@ -15,21 +15,22 @@
 from .common import setup_function, teardown_function
 
 test_defs = [
-    dict(kwargs=dict(),
+    dict(kwargs=dict(Writer=ascii.Basic),
```

Comment:
Good question and catch!

The answer is actually because the reading of these tables fails without a format specification being passed to the read.  Thanks for prompting me to remember that.  Once I write that down it clearly supports exactly your point.

So I will remove these two added `Writers=ascii.Basic` entries in the `kwargs` dict.  (the one above and the next one).  But this means we now have two more test cases that fail to round-trip.